src/anm/events.js

```diff
@@ -164,28 +164,31 @@ MouseEventsSupport.prototype.dispatch = function(event) {
         return true;
     }
 
-    var dispatchedChild;  // found the matching child inside
-    if (owner.inside(localEvent)) {
-        owner.reverseEach(function(child) {
-            if (child.isActive()) {
-                dispatchedChild = child.dispatchMouseEvent(localEvent) ? child : null;
-                if (dispatchedChild) return false; // stop iteration of reverseEach
-            }
-        });
+    var dispatchedByOwner, // handled event myself
+        dispatchedByChild; // not handled myself, but found the matching child inside
 
-        if (!dispatchedChild) {
-            if (localEvent.type === 'mousemove') {
-                this.processMove(localEvent); // fire mouseenter/mouseexit if required
-            }
-            if ((localEvent.type === 'mouseclick') || (localEvent.type === 'mousedown')) {
-                state.pressedNode = this.owner;
-            }
-            this.owner.fire(localEvent.type, localEvent);
+    // here and below localEvent has properties `.x` and `.y`, so duck typing works
+    dispatchedByOwner = owner.inBounds && owner.inBounds(localEvent) && owner.inside && owner.inside(localEvent);
```

Comment:
@skavish following [this conversation](https://github.com/Animatron/player/pull/383#issuecomment-175063710):

Actually, in case of mouse events, we need to check every element separately, I mean check if point belongs to the child _and this child dispatched the event_ (is that right, @kirillk?)  and continue iteration only if _both_ conditions are true. I [tried to start this way](https://github.com/Animatron/player/commit/7b3c09fca0b505a36d2419e69980cdeac225dcbc), with `inBounds` iterating through children but haven't found the good way to fit it in dispatching, the way you describe it, it needs to have some callback passed to `.inside` so it may stop iteration if child dispatched the event.